src/main/java/org/jfrog/hudson/release/PromotionUtils.java

```diff
@@ -20,83 +14,46 @@
     /**
      * Two stage promotion, dry run and actual promotion to verify correctness.
      *
-     * @param promotion
-     * @param client
-     * @param listener
-     * @param buildName
-     * @param buildNumber
-     * @throws IOException
      */
-    public static boolean promoteAndCheckResponse(Promotion promotion, ArtifactoryBuildInfoClient client, TaskListener listener,
-                                                  String buildName, String buildNumber) throws IOException {
+    public static boolean promoteAndCheckResponse(Promotion promotion, ArtifactoryManager artifactoryManager, TaskListener listener,
+                                                  String buildName, String buildNumber) {
         // If failFast is true, perform dry run first
-        HttpEntity entity = null;
         if (promotion.isFailFast()) {
             promotion.setDryRun(true);
             listener.getLogger().println("Performing dry run promotion (no changes are made during dry run) ...");
-            try (CloseableHttpResponse dryResponse = client.stageBuild(buildName, buildNumber, promotion)) {
-                entity = dryResponse.getEntity();
-                validatePromotionSuccessful(dryResponse, true, promotion.isFailFast(), listener);
+
+            try {
+                artifactoryManager.stageBuild(buildName, buildNumber, promotion);
+                listener.getLogger().println("Dry run finished successfully.\nPerforming promotion ...");
             } catch (IOException e) {
-                listener.error(e.getMessage());
-                return false;
-            } finally {
-                EntityUtils.consumeQuietly(entity);
+                if (onPromotionFailFast(true, promotion.isFailFast(), listener)) {
+                    return false;
+                }
+                listener.getLogger().println("Dry run failed to perform promotion ...");
```

Comment:
`failFast` is always true, therefore the log in line 32 is a dead code.
```suggestion
                onPromotionFailFast(true, true, listener);
                return false;
```